File: TaskBuilder/Models/Function/IInputModel.cs
using TaskBuilder.Functions;
using TaskBuilder.Models.Function.InputValue;

namespace TaskBuilder.Models.Function
{
    public interface IInputModel : IParameterModel
    {
        bool InlineOnly { get; }

        InputType InputType { get; }

        IInputValueModel StructureModel { get; }

        IInputValueModel FilledModel { get; }
    }
}

File: TaskBuilder.React/models/function/IInputModel.ts
import { IParameterModel } from "./IParameterModel";
import { InputType } from "./InputType";
import { IInputValueModel } from "./inputValue/IInputValueModel";

export interface IInputModel extends IParameterModel {
    inlineOnly: boolean;
    inputType: InputType;
    structureModel: IInputValueModel;
    filledModel: IInputValueModel;
}